src/index.js

```diff
@@ -30,35 +32,53 @@ const App = () => {
 
     useEffect(() => {
         localStorage.setItem(LOCAL_STORAGE_KEY_DATA, JSON.stringify(todos));
-    }, [todos]); // this effect only fires if the todos array changes: which is toggle complete and remove tasks
+        enableClearButton();
+    }, [todos]);
 
     const handleAddItem = () => {
         const currentInput = toDoInput.current.value;
         const pickedDueDate = dueDateInput.current.value;
-
-        if (currentInput && pickedDueDate){ // will implement validations in next step
-            const newTodoItem = {
-                id: uuidv4(),
-                name: currentInput,
-                due: pickedDueDate,
-                completed: false
-            };
-
-            setToDos([...todos, newTodoItem]);
-            toDoInput.current.value = null;
-            dueDateInput.current.value = null;
-        }
+        const newTodoItem = {
+            id: uuidv4(),
+            name: currentInput,
+            due: pickedDueDate,
+            completed: false
+        };
+
+        setToDos([...todos, newTodoItem]);
+        toDoInput.current.value = null;
+        dueDateInput.current.value = null;
+        setAddButton(true);
     }
 
     const handleCompleteItem = (itemID) => {
         const toToggleItem = todos.find(({ id }) => id === itemID);
         toToggleItem.completed = !toToggleItem.completed;
         setToDos([...todos]);
+        enableClearButton();
```

Comment:
Whenever the user check and uncheck a box, we need to decide if the clear button should be disabled.